File: src/main/scala/almoneya/Obligation.scala
package almoneya

import org.joda.time.LocalDate

case class Obligation(id: Option[ObligationId] = None,
                      account: Account,
                      description: Option[Description] = None,
                      startOn: LocalDate,
                      endOn: Option[LocalDate] = None,
                      amount: Amount,
                      every: Every,
                      period: Period) extends DueOnOrAfter {
    assert(endOn.isEmpty || endOn.get.isAfter(startOn))
}

File: lib/repositories/obligation.rb
module Repositories
  Obligation = Struct.new(:obligation_id, :envelope, :description, :every, :period, :start_on, :end_on, :amount, :created_at, :updated_at) do
    alias_method :id, :obligation_id

    def envelope_name
      envelope.name
    end
  end
end
